beps/0005-definitive-plugin-listing/README.md

```diff
@@ -0,0 +1,240 @@
+---
+title: Definitive Plugin Listing
+status: implementable
+authors:
+  - '@aramissennyeydd'
+owners:
+  - '@aramissennyeydd'
+project-areas:
+  - core
+creation-date: 2024-02-16
+---
+
+<!--
+**Note:** When your BEP is complete, all these pre-existing comments should be removed
+
+When editing BEPs, aim for tightly-scoped, single-topic PRs to keep discussions focused. If you disagree with what is already in a document, open a new PR with suggested changes.
+-->
+
+# BEP: Definitive Plugin Listing
+
+<!-- Before merging the initial BEP PR, create a feature issue and update the below link. You can wait with this step until the BEP is ready to be merged. -->
+
+[**Discussion Issue**](https://github.com/backstage/backstage/issues/23496)
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+- [Design Details](#design-details)
+- [Release Plan](#release-plan)
+- [Dependencies](#dependencies)
+- [Alternatives](#alternatives)
+
+## Summary
+
+<!--
+The summary of the BEP is a few paragraphs long and give a high-level overview of the features to be implemented. It should be possible to read *only* the summary and understand what the BEP is proposing to accomplish and what impact it has for users.
+-->
+
+The goal of this BEP is to define a new API that can be used to get a list of the currently installed features for an instance or a deployment of multiple different kinds of instances (called a split backend deployment). Plugins can then use this API to dynamically adjust their behavior, either tailoring it for a specific set of plugins or being able to exhaustively search across all installed plugins and/or modules.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this BEP. Describe why the change is important and the benefits to users.
+-->
+
+Split backends are a consistently difficult space to operate in and design for. There has been a growing desire for the framework to provide a way to get a list of the installed plugins. This was nearly impossible in the old backend, where plugins were hosted on denormalized routes and had non-standard startup sequences. In the new backend, this has become significantly more doable. Moving this forward would unblock a number of cases that require knowledge of your entire Backstage installation, namely a single OpenAPI spec for your instance, checking installed permissions, and DevTools information.
+
+### Goals
+
+<!--
+List the specific goals of the BEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+1. As an integrator, I can now get a list of currently installed features across my deployment.
+1. As an integrator/administrator, I can get a list of currently installed features for a single instance in my deployment.
```

Comment:
```suggestion
1. As a plugin builder, I can now get a list of currently installed features across my deployment.
1. As a plugin builder, I can get a list of currently installed features for a single instance in my deployment.
```

Right? Really the role that would be doing that. If you want to expose the available features for an integrator/operator you'd need to do that through a plugin.